server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -101,7 +103,29 @@ private static XContentBuilder mappings() {
         }
     }
 
-    public void putSynonymsset(String resourceName, SynonymsSet synonymsset, ActionListener<PutSynonymsAction.Response> listener) {
+    private static SynonymRule hitToSynonymRule(SearchHit hit) {
+        return new SynonymRule(hit.getId(), (String) hit.getSourceAsMap().get(SynonymRule.SYNONYMS_FIELD.getPreferredName()));
+    }
+
+    public void getSynonymsSet(String resourceName, int from, int size, ActionListener<SynonymsSetResult> listener) {
+        client.prepareSearch(SYNONYMS_ALIAS_NAME)
+            .setQuery(QueryBuilders.termQuery(SYNONYMS_SET_FIELD, resourceName))
+            .setFrom(from)
+            .setSize(size)
+            .execute(listener.delegateFailure((searchResponseListener, searchResponse) -> {
+                final long totalSynonymRules = searchResponse.getHits().getTotalHits().value;
+                if (totalSynonymRules == 0) {
```

Comment:
We should forbid creating a Synonyms Set with 0 elements, as we have no way of understanding whether it does not exist or it has 0 elements